packages/autocomplete-core/src/onInput.ts

```diff
@@ -29,6 +29,13 @@ interface OnInputParams<TItem extends BaseItem>
 
 const runConcurrentSafePromise = createConcurrentSafePromise();
 
+// All three values are used in the `then` callback, but because callbacks can
+// execute in a different order than the call order, the values could be stale.
```

Comment:
It always helps to give a scenario where it applies.

```suggestion
// All three values are used in the `then` callback, but because callbacks can
// execute in a different order than the call order (e.g., on an unstable network),
// the values could be stale.
```